rootfs/etc/nginx/lua/util/dns.lua

```diff
@@ -28,48 +27,65 @@ local function a_records_and_max_ttl(answers)
   return addresses, ttl
 end
 
-function _M.resolve(host)
-  local cached_addresses = cache:get(host)
-  if cached_addresses then
-    local message = string.format(
-      "addresses %s for host %s was resolved from cache",
-      table.concat(cached_addresses, ", "), host)
-    ngx.log(ngx.INFO, message)
-    return cached_addresses
-  end
-
+local function resolve_host(host, qtype)
   local r
   r, err = resolver:new{
-    nameservers = util.deepcopy(configuration.nameservers),
+    nameservers = configuration.nameservers,
     retrans = 5,
     timeout = 2000,  -- 2 sec
   }
```

Comment:
This should be done once in `.resolve` - you don't need to initialize different resolver per query.

Reply:
done